posts/2022-01-21-linking-metrics-and-traces-with-exemplars.adoc

```diff
@@ -0,0 +1,154 @@
+= Linking metrics and traces with Exemplars
+LeonorLunatech
+v1.0, 2022-01-21
+:title: Linking metrics and traces with Exemplars
+:imagesdir: ../media/2022-01-21-linking-metrics-and-traces-with-exemplars
+:lang: en
+:tags: [grafana, prometheus, exemplars, scala]
+
+What if you could easily link a metric with a trace? And what if I told you that you can do it visually?
+Meet Exemplars. They are displayed as diamonds in the graph shown above.
+
+By the end of this blog post you will be able to setup a complete docker-compose with Grafana, Tempo, Prometheus and a simple application that produces metrics and traces and links them with exemplars.
+
+== What are Exemplars
+But what are exemplars, exactly? Exemplars are a concept from https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#exemplars[Prometheus]
+"Exemplars are references to data outside of the MetricSet. A common use case are IDs of program traces."
+
+Grafana graphs support Prometheus exemplars. They are displayed as diamonds in the graph visualization. A mouse hovering, over the diamond, triggers a popup with details about the trace which, in turn, displays a button `Query  with tempo` which, when clicked, takes you to the corresponding trace visualization in Grafana Tempo.
```

Comment:
```suggestion
Grafana graphs support Prometheus exemplars. They are displayed as diamonds in the graph visualization. A mouse hovering over the diamond triggers a popup with details about the trace which, in turn, displays a button `Query with tempo` which, when clicked, takes you to the corresponding trace visualization in Grafana Tempo.
```

Reply:
I realised I forgot to push one modification 🤦  This sentence was already improved